src/SIMD.jl

```diff
@@ -121,7 +121,8 @@ end
 (::Type{Vec}){N,T<:ScalarTypes}(xs::NTuple{N,T}) = Vec{N,T}(xs)
 
 # Convert between vectors
-@inline Base.convert{N,T}(::Type{Vec{N,T}}, v::Vec{N}) = Vec{N,T}(Tuple(v))
+@inline Base.convert{N,T1, T2}(::Type{Vec{N,T1}}, v::Vec{N, T2}) = Vec{N,T1}(Tuple(v))
```

Comment:
I first did it like that but then the second function didn't get called at all. I would have thought the second function would be more specific but apparently not. I have had similar things like this happened before and just specifying all parameters usually solves it.